src/main/java/net/snowflake/ingest/streaming/internal/ParquetRowBuffer.java

```diff
@@ -208,16 +236,19 @@ int getTempRowCount() {
   @Override
   void reset() {
     super.reset();
-    data.clear();
+    createFileWriter();
   }
 
   @Override
-  public void close(String name) {
+  void closeInternal() {
     this.fieldIndex.clear();
-    logger.logInfo(
-        "Trying to close parquet buffer for channel={} from function={}",
-        channelFullyQualifiedName,
-        name);
+    if (data != null) {
+      try {
+        data.close();
+      } catch (IOException e) {
+        throw new SFException(ErrorCode.INTERNAL_ERROR, "Failed to close parquet writer", e);
+      }
```

Comment:
I am not 100% sure but otherwise `closeInternal()` will have to throw `IOException` and all callers have to handle it. I do not see much value in this and do not see a problem with wrapping it. Do you have a concern about wrapping? We will still see the cause in the stack trace.